src/abstraction-memory.c

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: LGPL-2.1-only
+/**
+ * Libcgroup abstraction layer for the memory controller
+ *
+ * Copyright (c) 2021-2025 Oracle and/or its affiliates.
+ * Author: Tom Hromatka <tom.hromatka@oracle.com>
+ */
+
+#include "abstraction-common.h"
+#include "abstraction-map.h"
+
+#include <libcgroup.h>
+#include <libcgroup-internal.h>
+
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <errno.h>
+#include <stdio.h>
+
+#define V1_MAX_STR "9223372036854771712"
+#define V2_MAX_STR "max"
+
+#define LL_MAX 8192
```

Comment:
The maximum allowed value range is of 20 characters and you may want to reduce the array size here to reduce the unused bytes.